src/Sylius/Behat/spec/Context/Ui/Admin/ManagingTaxCategoryContextSpec.php

```diff
@@ -103,4 +108,282 @@ function it_throws_an_exception_if_the_message_on_a_page_is_not_related_to_delet
+    {
+        $taxCategoryCreatePage->create()->shouldBeCalled();
+
+        $this->iAddIt();
+    }
+
+    function it_asserts_if_a_resource_was_successfully_created(
+        UpdatePageInterface $taxCategoryUpdatePage,
+        TaxCategoryInterface $taxCategory
+    ) {
+        $taxCategoryUpdatePage->isOpen()->shouldBeCalled();
+        $taxCategoryUpdatePage->hasResourceValues([
+            'name' => 'Food and Beverage',
+            'code'=> 'food_and_beverage',
+            'description' => 'Best stuff to get wasted in town',
+        ])->willReturn(true);
+
+        $taxCategory->getCode()->willReturn('food_and_beverage');
+        $taxCategory->getName()->willReturn('Food and Beverage');
+        $taxCategory->getDescription()->willReturn('Best stuff to get wasted in town');
+
+        $this->thisTaxCategoryShouldAppearInTheRegistry($taxCategory);
+    }
+
+    function it_throws_an_exception_if_resource_does_not_have_proper_fields_filled(
+        UpdatePageInterface $taxCategoryUpdatePage,
+        TaxCategoryInterface $taxCategory
+    ) {
+        $taxCategoryUpdatePage->isOpen()->shouldBeCalled();
+        $taxCategoryUpdatePage->hasResourceValues([
+            'name' => 'Food and Beverage',
+            'code'=> 'food_and_beverage',
+            'description' => 'Best stuff to get wasted in town',
+        ])->willReturn(false);
+
+        $taxCategory->getCode()->willReturn('food_and_beverage');
+        $taxCategory->getName()->willReturn('Food and Beverage');
+        $taxCategory->getDescription()->willReturn('Best stuff to get wasted in town');
+
+        $this
+            ->shouldThrow(new \InvalidArgumentException('Tax category with code food_and_beverage was found, but fields are not assigned properly'))
+            ->during('thisTaxCategoryShouldAppearInTheRegistry', [$taxCategory])
+        ;
+    }
+
+    function it_checks_if_a_resource_was_successfully_created(NotificationAccessorInterface $notificationAccessor)
+    {
+        $notificationAccessor->hasSuccessMessage()->willReturn(true);
+        $notificationAccessor->isSuccessfullyCreatedFor('tax_category')->willReturn(true);
+
+        $this->iShouldBeNotifiedAboutSuccessfulCreation();
+    }
+
+    function it_throws_an_exception_if_the_creation_page_does_not_have_success_message(NotificationAccessorInterface $notificationAccessor)
+    {
+        $notificationAccessor->hasSuccessMessage()->willReturn(false);
+
+        $this
+            ->shouldThrow(new \InvalidArgumentException('Message type is not positive'))
+            ->during('iShouldBeNotifiedAboutSuccessfulCreation', [])
+        ;
+    }
+
+    function it_throws_an_exception_if_the_message_on_a_page_is_not_related_to_creation(NotificationAccessorInterface $notificationAccessor)
+    {
+        $notificationAccessor->hasSuccessMessage()->willReturn(true);
+        $notificationAccessor->isSuccessfullyCreatedFor('tax_category')->willReturn(false);
```

Comment:
I'm not sure which approach is better. I'm not convinced about using const in specs. I jus prefer to write it by my self, especially, because I do not have this const when I am writing first lines of spec